### DISCUSSION TITLE
using the useSigner hook correctly?

### DISCUSSION THREAD

**Discussion by Dope-Ass-Ninja** [wevm/wagmi#845] [Q&A] [answered]
Hello everyone! I've really enjoyed using wagmi as opposed to some other react hook libraries and I had a question relating to useSigner hooks & how to use them.

For context:
 I am creating a dApp for minting an ERC-721 contract through Immutable X (IMX) and their API. I am using wagmi for the L1 side, and will need to either Connect to or Create a Stark Wallet (L2) for the IMX side. When a user connects their account, I am currently attempting to use the useSigner wagmi hook to supply the correct signer to the generateStarkWallet function supplied by IMX. This function takes in an L1 signer as a parameter and outputs the user's connected L2 wallet or prompts to create a new wallet in the response. 

The issue:
When loading the page and connecting your wallet, the values of objects like `{ data: Signer }` in useSigner() are null. I've tried console.logging these values in the onSuccess or onSettled methods and get the same result of null on page load / when auto connecting to an address. I suspect the issue may be related to how I'm calling the function in this useEffect hook so I'll include that as well: 
```
const { data: Signer, isError, isLoading } = useSigner();

useEffect(() => {
      const generateL2Wallet = async () => {
        const starkWallet = await generateStarkWallet(Signer);
        console.log(starkWallet);
      }
      if(address && isConnected) {
        console.log(Signer);
        console.log(isConnected);
        generateL2Wallet();
      }
    }, [address])
```

Any insight to this would be greatly appreciated! I'm not sure if there's a way to lazy load until `address` is connected, but it's seeming like a timing issue right now.

**Comment by tmm** [comment, marked as answer]
Looks like you're missing `Signer` in the dependency array.

**Reply by Dope-Ass-Ninja** [reply]
Thanks for the quick response, I've just started development in the last year or so and can definitely make silly mistakes often!
